admin/class-botamp-admin.php

```diff
@@ -361,6 +404,241 @@ public function entity_url_cb() {
 		echo $this->print_field_select( 'entity_url' );
 	}
 
+	public function add_messenger_widget( $checkout ) {
+		if ( $this->get_option( 'order_notifications' ) !== 'enabled' ) {
+			return;
+		}
+
+		$page_attributes = $this->botamp->me->get()->getBody()['data']['attributes'];
+
+		$ref = uniqid( 'botamp_' . $_SERVER['HTTP_HOST'] . '_', true );
+
+		echo '<input type="hidden" name="botamp_contact_ref" value="' . $ref . '">';
+
+		echo '<div id="notifications"><h3>' . __( 'Notifications' ) . '</h3>';
+		echo "<script>
+			 	window.fbAsyncInit = function() {
+				    FB.init({
+				      appId      : '{$page_attributes['facebook_app_id']}',
+				      xfbml      : true,
+				      version    : 'v2.6'
+				    });
+				};
+
+			  	(function(d, s, id){
+				    var js, fjs = d.getElementsByTagName(s)[0];
+				      if (d.getElementById(id)) {return;}
+				      js = d.createElement(s); js.id = id;
+				      js.src = '//connect.facebook.net/en_US/sdk.js';
+				      fjs.parentNode.insertBefore(js, fjs);
+				  }(document, 'script', 'facebook-jssdk')
+			  	);
+			</script>
+			<div class='fb-send-to-messenger'
+			  messenger_app_id='{$page_attributes['facebook_app_id']}'
+			  page_id='{$page_attributes['facebook_id']}'
+			  data-ref='$ref'
+			  color='blue'
+			  size='standard'>
+			</div></div>";
+
+	}
```

Comment:
Why is this method and all the other ones which hook into the frontend in the admin class?